Tests.SimpleConfigSections/ConfigurationSectionBasedOnClassSpecs.cs

```diff
@@ -64,16 +84,41 @@ public override object  ConvertFrom(ITypeDescriptorContext context, System.Globa
         }
     }
 
+    public enum EnumWithDefault
+    {
+        Default = 0,
+        Other
+    }
+
+    public enum EnumWithNoDefault
+    {
+        First = 1,
+        Second = 2
+    }
+
     public class SimpleConfiguration
     {
-		[Display(Name = "name")]
+        [Display(Name = "name")]
         public virtual string Name { get; set; }
         public virtual int Count { get; set; }
 
         public virtual SimpleConfiguration Child { get; set; }
 
         public virtual Regex Matcher { get; set; }
 
+        public virtual EnumWithDefault Enum1 { get; set; }
+        [Default(DefaultValue = EnumWithDefault.Other)]
+        public virtual EnumWithDefault Enum2 { get; set; }
+
+        public virtual EnumWithNoDefault Enum3 { get; set; }
+        [Default(DefaultValue = EnumWithNoDefault.Second)]
+        public virtual EnumWithNoDefault Enum4 { get; set; }

```

Comment:
Well, I am terrible at naming.. feel free to try any other combination while merging it. heh ;)